frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -154,17 +162,97 @@ function FileManagerHistory:onShowHist()
         covers_fullscreen = true, -- hint for UIManager:_repaint()
         is_borderless = true,
         is_popout = false,
+        title_bar_left_icon = "appbar.menu",
+        onLeftButtonTap = function() self:showHistDialog() end,
         onMenuHold = self.onMenuHold,
         onSetRotationMode = self.MenuSetRotationModeHandler,
         _manager = self,
     }
 
     self:updateItemTable()
     self.hist_menu.close_callback = function()
+        self.statuses_fetched = nil
+        self.filter = nil
         UIManager:close(self.hist_menu)
     end
     UIManager:show(self.hist_menu)
     return true
 end
 
+function FileManagerHistory:showHistDialog()
+    if not self.statuses_fetched then
+        local _status
```

Comment:
(Btw, no need for an underscore for a local variable, it's always private :) and removing it would make it clearer and avoid typos (forgetting it) next time you update this bit.)